implementations/elixir/ockam/ockam/test/ockam/secure_channel/encrypted_transport_protocol/aead_aes_gcm_test.exs

```diff
@@ -88,4 +88,50 @@ defmodule Ockam.SecureChannel.EncryptedTransportProtocol.AeadAesGcmTests do
     {:ok, ciphertext, _encryptor} = Encryptor.encrypt(<<>>, plain, encryptor)
     {:ok, ^plain, _decryptor} = Decryptor.decrypt(<<>>, ciphertext, decryptor)
   end
+
+  test "out of order, exact sliding window" do
+    # We can't share the _same_ k between encryptor and decryptor on the same vault, as when the encryptor
+    # rotate the key, it destroy the old k.  But that might still be used by the decryptor to decrypt yet-to-be
+    # delivered packets.
+    {:ok, encryptor_vault} = SoftwareVault.init()
+    {:ok, decryptor_vault} = SoftwareVault.init()
+    shared_k = :crypto.strong_rand_bytes(32)
+    {:ok, ke} = Vault.secret_import(encryptor_vault, [type: :aes], shared_k)
+    {:ok, kd} = Vault.secret_import(decryptor_vault, [type: :aes], shared_k)
+    encryptor = Encryptor.new(encryptor_vault, ke, 0, 32)
+    decryptor = Decryptor.new(decryptor_vault, kd, 0, 32)
+
+    {msgs, _encryptor} =
+      Enum.reduce(0..100, {[], encryptor}, fn _i, {acc, encryptor} ->
+        plain = :crypto.strong_rand_bytes(64)
+        {:ok, ciphertext, encryptor} = Encryptor.encrypt(<<>>, plain, encryptor)
+        {[{plain, ciphertext} | acc], encryptor}
+      end)
+
+    msgs = msgs |> Enum.reverse()
+
+    # msg is out of the receive window
+    {_, ciphertext} = Enum.at(msgs, 33)
+    {:error, :out_of_window} = Decryptor.decrypt(<<>>, ciphertext, decryptor)
+
+    # msg is on the receive window
+    {plaintext, ciphertext} = Enum.at(msgs, 10)
+    {:ok, ^plaintext, decryptor} = Decryptor.decrypt(<<>>, ciphertext, decryptor)
+
+    # msg is on the receive window
+    {plaintext, ciphertext} = Enum.at(msgs, 2)
+    {:ok, ^plaintext, decryptor} = Decryptor.decrypt(<<>>, ciphertext, decryptor)
+
+    # msg is on the receive window now
+    {plaintext, ciphertext} = Enum.at(msgs, 33)
```

Comment:
It would be good to add more boundary tests:

 - 1 before the lower part of the window
 - exactly on the lower part of the window
 - 1 after the lower part of the window
 - 1 before the upper part of the window
 - exactly on the upper part of the window
 - 1 after the upper part of the window

Reply:
thanks,  I now adapted  the test in [rust side](https://github.com/build-trust/ockam/blob/344c4736e31c3b104bdc1a31f133603b4377138e/implementations/rust/ockam/ockam_identity/src/secure_channel/nonce_tracker.rs#L61-L83) so we use the same case on both